apps/betterangels/src/app/(private-screens)/add-note/DateAndTime/index.tsx

```diff
@@ -0,0 +1,143 @@
+  UpdateNoteMutationVariables,
+} from '@monorepo/expo/betterangels';
+import { Regex } from '@monorepo/expo/shared/static';
+import {
+  DatePicker,
+  FieldCard,
+  TextMedium,
+  TextRegular,
+} from '@monorepo/expo/shared/ui-components';
+import { debounce } from '@monorepo/expo/shared/utils';
+import { format, setHours, setMinutes } from 'date-fns';
+import { RefObject, useRef, useState } from 'react';
+import { ScrollView, View } from 'react-native';
+
+interface IDateAndTimeProps {
+  expanded: string | undefined | null;
+  setExpanded: (expanded: string | undefined | null) => void;
+  scrollRef: RefObject<ScrollView>;
+  interactedAt: Date;
+  noteId: string | undefined;
+}
+
+type TDateAndTime = {
+  date: Date;
+  time: Date;
+};
+
+const endOfDay = new Date(new Date().setHours(23, 59, 59, 999));
+
+export default function DateAndTime(props: IDateAndTimeProps) {
+  const { expanded, setExpanded, scrollRef, interactedAt, noteId } = props;
+
+  const [updateNote] = useMutation<
+    UpdateNoteMutation,
+    UpdateNoteMutationVariables
+  >(UPDATE_NOTE);
+
+  const [dateTime, setDateTime] = useState<TDateAndTime>({
+    date: interactedAt,
+    time: interactedAt,
+  });
+
+  const noteRef = useRef(dateTime);
+  const isDateAndTime = expanded === 'Date And Time';
+
+  const updateNoteFunction = useRef(
+    debounce(async (key: 'time' | 'date', value: string | Date) => {
+      if (!noteId || !value) return;
+      const currentNote = noteRef.current;
+      const dateValue = key === 'date' ? value : new Date(currentNote.date);
+      const timeValue = key === 'time' ? value : new Date(currentNote.time);
+      let updatingField = value;
+
+      if (timeValue instanceof Date && dateValue instanceof Date) {
+        const hours = timeValue.getHours();
+        const minutes = timeValue.getMinutes();
+        const combinedDateTime = setMinutes(
+          setHours(dateValue, hours),
+          minutes
+        );
+
+        updatingField = new Date(combinedDateTime).toISOString();
+      } else {
+        throw new Error('Both timeValue and dateValue should be Date objects');
+      }
+
+      try {
+        await updateNote({
+          variables: {
+            data: {
+              id: noteId,
+              interactedAt: updatingField,
+            },
+          },
+        });
+      } catch (err) {
+        console.log(err);
+      }
+    }, 500)
+  ).current;
+
+  const onChange = (key: 'date' | 'time', value: Date) => {
+    setDateTime({ ...dateTime, [key]: value });
+    updateNoteFunction(key, value);
+  };
+
+  return (
+    <FieldCard
+      scrollRef={scrollRef}
+      expanded={expanded}
+      mb="xs"
+      setExpanded={() => setExpanded(isDateAndTime ? null : 'Date And Time')}
+      title="Date And Time"
+      actionName={
+        !dateTime.date && !isDateAndTime ? (
+          <TextMedium size="sm">Add Date and Time</TextMedium>
+        ) : dateTime.date && !isDateAndTime ? (
+          <TextRegular mb="md">
```

Comment:
```suggestion
          <TextMedium size="xxs">Add Date and Time</TextMedium>
        ) : dateTime.date && !isDateAndTime ? (
          <TextRegular mb="xxs" size="sm">
```
**before** -> **after**

<img width="276" alt="image" src="https://github.com/user-attachments/assets/b0b32fab-3ebf-485c-9ec9-dc0be33b33a4">
<img width="275" alt="image" src="https://github.com/user-attachments/assets/2eeb6f0e-ccef-47d8-a40e-d0a67d243262">

Reply:
updated to 

<TextMedium>
just saw location is same, removed mb and size for both "sm"